pallets/automation-time/src/lib.rs

```diff
@@ -589,6 +591,36 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Cancel task by schedule_as
+		///
+		/// # Parameters
+		/// * `schedule_as`: The schedule_as account of the task.
+		/// * `task_id`: The id of the task.
+		///
+		/// # Errors
+		/// * `TaskDoesNotExist`: The task does not exist.
+		/// * `TaskScheduleAsNotMatch`: The schedule_as account of the task does not match.
+		#[pallet::call_index(8)]
+		#[pallet::weight(<T as Config>::WeightInfo::cancel_task_with_schedule_as_full())]
+		pub fn cancel_task_with_schedule_as(
+			origin: OriginFor<T>,
+			owner_id: AccountOf<T>,
+			task_id: TaskIdV2,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let task = AccountTasks::<T>::get(owner_id, task_id.clone())
+				.ok_or(Error::<T>::TaskDoesNotExist)?;
+
+			if !matches!(task.clone().action, Action::XCMP { schedule_as: Some(ref s), .. } if s == &who)
+			{
+				return Err(Error::<T>::TaskScheduleAsNotMatch.into())
+			}
+
+			Self::remove_task(task_id, task);
```

Comment:
I mentioned that we need to TODO here but it actually should be in the remove_task wrapper(), so no action here.

The TODO is about refund reserved execution fees.

Reply:
Fixed